library/lib.sh

```diff
@@ -250,6 +255,104 @@ rolesRunPlaybook() {
     rolesUploadLogs "$LOGFILE"
 }
 
+rolesCS8InstallPython() {
+    # Install python on managed node when running CS8 with ansible!=2.9
+    if [ "$ANSIBLE_VER" != "2.9" ] && grep -q 'CentOS Stream release 8' /etc/redhat-release; then
+        rlRun "dnf install -y python$PYTHON_VERSION"
+    fi
+}
+
+rolesDistributeSSHKeys() {
+    # name: Distribute SSH keys when provisioned with how=virtual
+    local tmt_tree_provision=$1
+    local control_node_id_ecdsa_pub=$tmt_tree_provision/control_node/id_ecdsa.pub
+    if [ -f "$control_node_id_ecdsa_pub" ]; then
+        cat "$control_node_id_ecdsa_pub" >> ~/.ssh/authorized_keys
+    fi
+}
+
+rolesEnableHA() {
+# This function enables the ha repository on platforms that require it and do not have it enabled by default
+# The ha repository is required by the mssql and ha_cluster roles
+    local ha_reponame
+    if rlIsRHELLike 7; then
+        return
+    fi
+    if rlIsRHELLike 8; then
+        ha_reponame=ha
+    elif rlIsRHELLike ">8"; then
+        ha_reponame=highavailability
+    fi
+    rlRun "dnf config-manager --set-enabled $ha_reponame"
+}
+
+rolesDisableNFV() {
+    # The nfv-source repo causes troubles in CentOS-9 Stream compose while system-roles testing
+    if [ "$(find /etc/yum.repos.d/ -name 'centos-addons.repo' | wc -l )" -gt 0 ]; then
+        rlRun "sed -i '/^\[nfv-source\]/,/^$/d' /etc/yum.repos.d/centos-addons.repo"
+    fi
+}
+
+rolesGenerateTestDiscs() {
+# This function generates test disks from provision.fmf
+# This is required by storage and snapshot roles
+    local role_path=$1
+    local provisionfmf="$role_path"/tests/provision.fmf
+    local -i i=0
+    local disk_provisioner_dir TARGETCLI_CMD disks disk file
+    rolesCloneRepo "$role_path"
+    if [ ! -f "$provisionfmf" ]; then
+        rlRun "rm -rf ${role_path}"
+        return
+    fi
+    if ! grep -q drive: "$provisionfmf"; then
+        rlRun "rm -rf ${role_path}"
+        return
+    fi
+    if type -p python3; then
+        PYTHON=python3
+    elif type -p python; then
+        PYTHON=python
+    elif type -p python2; then
+        PYTHON=python2
```

Comment:
```bash
    elif  [ -f /usr/libexec/platform-python ]; then
        PYTHON=/usr/libexec/platform-python
```

Reply:
What's this for?

Reply:
It is another place to look for a python interpreter - most RHEL have this as the "default" python interpreter.